lib/Dialect/TTIR/IR/TTIROps.cpp

```diff
@@ -1012,6 +1012,39 @@ ::mlir::LogicalResult mlir::tt::ttir::AllGatherOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// AllReduceOp
+//===----------------------------------------------------------------------===//
+
+// AllReduceOp verification
+::mlir::LogicalResult mlir::tt::ttir::AllReduceOp::verify() {
+  ::mlir::RankedTensorType inputType =
+      mlir::cast<RankedTensorType>(getInputs().front().getType());
+  int32_t dim = getDim();
+
+  if (dim >= inputType.getRank()) {
+    return emitOpError("Invalid dimension for all_reduce op.");
+  }
+
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// MeshShardOp
+//===----------------------------------------------------------------------===//
+
+// MeshShardOp verification
+::mlir::LogicalResult mlir::tt::ttir::MeshShardOp::verify() {
+  int32_t shardType = getShardType();
+
+  // currently we are only supporting replicate or devices from StableHLO
+  if (shardType != 1 && shardType != 3) {
```

Comment:
`kShardTypeReplicated` and `kShardTypeDevices`.  Perhaps we can have concrete enums defined in the op class definition for these?